classes/form/edit_conditions.php

```diff
@@ -250,6 +268,8 @@ public function validation($data, $files) {
                     $errors["repeatvalue[$value]"] = get_string('formexperimentvalueerror', 'tool_abconfig');
                 }
                 // Increment total and check value.
+                // If an empty string is returned, set value to 0.
+                $data['repeatvalue'][$value] = $data['repeatvalue'][$value] ? '' : 0;
```

Comment:
This says "If `$data['repeatvalue'][$value]` is true, set it to an empty string, else set it to 0".

I think you mean something like
```php
if ($data['repeatvalue'][$value] == '') { 
    $data['repeatvalue'][$value] = 0
}
```

Reply:
This has been fixed so ready for re-review please @marxjohnson